test/unit/middleware.js

```diff
@@ -23,9 +28,105 @@ describe('Middleware', () => {
 
       middleware.getUrl(fakeRequest, {}, () => {});
 
-      assert.equal(
-        fakeRequest.urlForGp,
-        'http://test/123456?apikey=secret');
+      expect(fakeRequest.urlForGp).to.equal('http://test/123456?apikey=secret');
+    });
+  });
+  describe('getGpDetails', () => {
+    const fakeRequest = {
+      urlForGp: 'http://test/test',
+    };
+    it('should return populated gp details for the practice', (done) => {
+      nock('http://test')
+        .get('/test')
+        .reply(200, getSampleResponse('gp_practice_by_ods_code'));
+
+      middleware.getDetails(fakeRequest, {}, () => {
+        expect(fakeRequest.gpDetails).to.be.an('object');
+        expect(fakeRequest.gpDetails).to.have.keys(['name', 'address', 'overviewLink']);
+        expect(fakeRequest.gpDetails.address).to.have.keys(
+          ['line1', 'line2', 'line3', 'line4', 'postcode']);
+        done();
+      });
+    });
+    it('should handle GP overview resource page not found', (done) => {
+      nock('http://test')
+        .get('/test')
+        .reply(404);
+
+      middleware.getDetails(fakeRequest, {}, (err) => {
+        expect(err.message).to.equal('GP Not Found');
+        expect(err.status).to.equal(404);
+        done();
+      });
+    });
+    it('should handle server error', (done) => {
+      nock('http://test')
+        .get('/test')
+        .reply(500);
+
+      middleware.getDetails(fakeRequest, {}, (err) => {
+        expect(err).to.equal('Error: 500');
+        done();
+      });
+    });
+    it('should handle http error', (done) => {
+      nock('http://test')
+        .get('/test')
+        .replyWithError('Bad HTTP stuff happened');
+
+      middleware.getDetails(fakeRequest, {}, (err) => {
+        expect(err.message).to.equal('Bad HTTP stuff happened');
```

Comment:
Mmh, not sure I fully understand or perhaps I do but I don't see it the same way. My take on it is...I don't really care what the error message is and don't want a typo to cause a test to fail. I want the test to fail if the error handling logic is incorrect/changes. Having the error message in a variable still means the test will be failing until such a time as the logic has been implemented. At least that is my understanding. Perhaps a voice/video conversation is required?